websurvey/lib/typewheels/form.js

```diff
@@ -140,6 +142,53 @@ function filterFields(form) {
   return form.fields.filter(field => field.type !== "thankyou_screen");
 }
 
+function _splitUrls(s) {
+  const re = RegExp(/https?:\/\/[^\s]+/gi);
+  const nonMatches = s.split(re).map(s => ["text", s]);
+  return nonMatches;
+}
+
+function getDynamicValue(qa, title) {
+  const regex = /:(.*)}}/;
+  const match = title.match(regex) ? title.match(regex) : false;
+  const ref = match ? match[1] : false;
+
+  const fieldValue = getFieldValue(qa, ref);
+
+  if (!match) {
+    return false;
+  }
+
+  if (!fieldValue || fieldValue === " ") {
+    throw new TypeError(
+      `Trying to interpolate a non-existent field value: ${title}`
+    );
+  }
+
+  return fieldValue;
+}
+
+function _interpolate(qa, title) {
+  const titledParsed = Mustache.parse(title);
+
+  title = titledParsed
+    .map(t =>
+      t[0] === "name" ? t[1].replace(t[1], getDynamicValue(qa, title)) : t[1]
+    )
+    .join("");
+
+  return title;
+}
+
+function interpolateField(qa, field) {
+  const keys = ["title"];
```

Comment:
Hrm, so we don't use "properties.description" here, in the web survey, but would it hurt to have it here (properties will still have a description, no?). Generally, we are trying to _abstract_ this into a shared library, so ideally we share the most possible!

Reply:
Good point, I've been thinking about things inversely only abstracting what is needed for the web survey so will go back and check the `form` and `validator` modules to see if I missed anything that could be included